Marlin/Marlin_main.cpp

```diff
@@ -5956,7 +5962,8 @@ inline void gcode_M17() {
           nozzle_timed_out |= thermalManager.is_heater_idle(e);
 
       #if ENABLED(ULTIPANEL)
-        if (nozzle_timed_out) ensure_safe_temperature();
+        if (nozzle_timed_out)
+          lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_CLICK_TO_HEAT_NOZZLE);
```

Comment:
Where is the logic to wait for the click, turn on the heater, and so on? I can't see how this is supposed to work.